nixio/cmd/nixexplore.py

```diff
@@ -0,0 +1,623 @@
+    second_dim_label, second_dim_unit = get_dim_label_and_unit(dimensions[1])
+
+    numeric_1st_dim_ticks = isinstance(first_dim_ticks[0], (int, float))
+    numeric_2nd_dim_ticks = isinstance(second_dim_ticks[0], (int, float))
+    numeric_data = isinstance(data[0, 0], (int, float))
+    data_conv_func = (lambda x: format % x) if numeric_data else str
+    dim_ticks_conv_func1 = (lambda x: format % x) if numeric_1st_dim_ticks else str
+    dim_ticks_conv_func2 = (lambda x: format % x) if numeric_2nd_dim_ticks else str
+
+    max_tick_len = max([len(dim_ticks_conv_func1(first_dim_ticks[-1])), len(first_dim_label)])
+    print("# data label: %s" % label)
+    print("# data unit: %s\n" % unit)
+    padding = " " * (max_tick_len - (len(first_dim_label) if first_dim_unit else 0))
+    print("# %s%s%s" % (first_dim_label, padding, second_dim_label))
+    padding = " " * (max_tick_len - (len(first_dim_unit) if first_dim_unit else 0))
+    print("# %s%s%s" % (first_dim_unit, padding, second_dim_unit))
+    # first line contains 2nd dim ticks
+    print(" " * max_tick_len + "   " + (" " * max_tick_len + "  ").join(map(dim_ticks_conv_func2, second_dim_ticks)))
+    # now dump the rest
+    for i in range(len(first_dim_ticks)):
+        print(dim_ticks_conv_func1(first_dim_ticks[i]) + "    " + "   ".join(map(data_conv_func, data[i, :])))
+        if i % 500 == 0:
+            progress(i, data.shape[0], status='Dumping ...')
+    print(end)
+
+
+def dump_threed(data, dimensions, label, unit, format="%.6f", end="\n\n"):
+    assert(len(data.shape) == 3 and len(dimensions) == 3)
+    ticks = get_ticks(dimensions[2], data.shape[2])
+    dim_label, dim_unit = get_dim_label_and_unit(dimensions[2])
+
+    for i in range(data.shape[2]):
+        print("# data[:, :, %i]: %s" % (i, dim_label + "%s%s" % (ticks[i], dim_unit) if dim_unit else ""))
+        dump_twod(data[:, :, i], [dimensions[0], dimensions[1]], label, unit, format, end="\n")
+
+    print(end)
+
+
+def dump_data_array(array, filename):
+    print("# File: %s\n# entity: %s\n# type: %s\n# id: %s" % (filename, array.name, array.type, array.id))
+    print("# created at: %s\n# last edited at: %s\n" %
+          (str(dt.datetime.fromtimestamp(array.created_at)),
+           str(dt.datetime.fromtimestamp(array.updated_at))))
+    dims = len(array.shape)
+    data = array[:]
+    if dims == 1:
+        dump_oned(data, array.dimensions[0], array.label, array.unit)
+    elif dims == 2:
+        dump_twod(data, array.dimensions, array.label, array.unit)
+    elif dims == 3:
+        dump_threed(data, array.dimensions, array.label, array.unit)
+    else:
+        print("Sorry, cannot dump data with more than 3 dimensions!")
+
+
+def data_dump(filename, arguments):
+    nix_file = open_nix_file(filename)
+    entities = find_data_entity(nix_file, arguments)
+    for e in entities:
+        if isinstance(e, nix.pycore.data_array.DataArray):
```

Comment:
```suggestion
        if isinstance(e, nix.DataArray):
```

`pycore` is gone.  Classes are exposed at the top level.